libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -527,6 +528,34 @@ private boolean isEvidenceVc(VcStoreItem item) throws CredentialParseException {
         return false;
     }
 
+    private boolean isNonZeroInt(JsonNode node) {
+        return node.isInt() && node.asInt() != 0;
+    }
+
+    private boolean isValidEvidence(VcStoreItem item) {
+        try {
+            JsonNode vcEvidenceNode = getVCClaimNode(item.getCredential(), VC_EVIDENCE);
+            return ((vcEvidenceNode.size() > 0)
+                    && isNonZeroInt(vcEvidenceNode.get(0).path(VC_EVIDENCE_VALIDITY))
+                    && isNonZeroInt(vcEvidenceNode.get(0).path(VC_EVIDENCE_STRENGTH)));
+        } catch (CredentialParseException e) {
+            return false;
+        }
+    }
+
+    public String getCredentialIssuerIfSingleValidEvidence(List<VcStoreItem> vcStoreItems) {
```

Comment:
This seems like it might be a bit over-specialised as a function. It's only returning a credential issuer because that's what the calling code in this case needs. It might be more re-usable if it just returned a single evidence item and the caller then called `.getCredentialIssuer()` on the return item.

Reply:
We added a new property name 'requiredAdditionalEvidence' under CRI in our config param files. Since we only need credential to access this property, that's what is being retuned. Yes, you are right, this method can be used when needed. I am not sure whether we should do tis now or when needed.